bootstrap/terraform/vm.tf

```diff
@@ -0,0 +1,216 @@
+terraform {
+  required_providers {
+    azurerm = {
+      source  = "hashicorp/azurerm"
+      version = "3.29.1"
+    }
+    azuread = {
+      source  = "hashicorp/azuread"
+      version = "2.30.0"
+    }
+    http = {
+      source  = "hashicorp/http"
+      version = "3.2.0"
+    }
+  }
+}
```

Comment:
Should we specify a minimum version instead of a specific version?
https://developer.hashicorp.com/terraform/language/providers/requirements#version-constraints
`version = ">= 3.29.1"` etc.

Since this is for local development and possibly CI steps, and not to upgrade production environments, I'd be fine with always using the latest versions personally.

Reply:
Ya that's a good point. It'd be nice if these could just use whatever the latest version is so that we don't need to worry about upgrading them over time. I'll use the `>=` 👍

Reply:
Done in [3d2cbee](https://github.com/hashicorp/vault-plugin-auth-azure/pull/65/commits/3d2cbeee231ccba733f24eefb79a8d12c73cb822).

Reply:
If you omit the required_providers block entirely terraform init will use the latest version at all times.

It's a trade-off between reproduceability/stability & toil of managing versions. 

I think for customer facing infrastructure changes having set version is a no brainer for safety. But for QoL-projects where a failed run is not a big deal I'm not against removing this block completely so we catch our own regressions.

Reply:
+1 @maxcoulombe good points

Reply:
+1 - I'll remove the required_provides block entirely. This will be consistent with what's in GCP secrets/auth too.

Reply:
Done in [cd179cd](https://github.com/hashicorp/vault-plugin-auth-azure/pull/65/commits/cd179cdc859e7e7aa5dc124afe038562930399f9).